backend/src/main.cpp

```diff
@@ -149,74 +150,51 @@ Napi::Boolean AddFrequency(const Napi::CallbackInfo& info)
         TRACK_LOG_WARNING("Could not add frequency, it already exists: {} {}", frequency, callsign);
         return Napi::Boolean::New(info.Env(), false);
     }
-    mClient->SetRx(frequency, false);
-    mClient->SetRadioGainAll(UserSession::currentRadioGain);
 
-    MainThreadShared::mApiServer->handleAFVEventForWebsocket(
-        sdk::types::Event::kFrequencyStateUpdate, {}, {});
+    RadioState newState;
+
+    newState.frequency = frequency;
+    newState.rx = false;
+    newState.tx = false;
+    newState.xc = false;
+    newState.headset = true;
+    newState.xca = false;
 
-    return Napi::Boolean::New(info.Env(), true);
+    auto result = RadioHelper::SetRadioState(MainThreadShared::mApiServer.get(), newState);
```

Comment:
Use the new shared method for setting state.

Reply:
I'd rather you you don't access the raw ptr behind the shared_ptr, if it's currently a unique_ptr change it to a shared_ptr and pass it by reference, that's the whole point of smart pointers